gbpservice/neutron/tests/unit/nfp/configurator/lib/test_schema_validator.py

```diff
@@ -0,0 +1,204 @@
+                  'provider_cidr': '11.0.0.0/24',
+                  'provider_interface_position': '2',
+                  'stitching_ip': '33.0.0.4',
+                  'stitching_cidr': '33.0.0.0/24',
+                  'stitching_interface_position': '3',
+                  'provider_mac': 'e1:6d:af:23:b8:91',
+                  'stitching_mac': 'e1:6d:af:23:b8:11',
+                  'rule_info': {'active_provider_mac': 'e1:6d:af:23:b8:91',
+                                'active_stitching_mac': 'e1:6d:af:23:b8:11',
+                                'active_fip': '192.168.100.141',
+                                'service_id':
+                                    '6350c0fd-07f8-46ff-b797-62acd2371234',
+                                'tenant_id':
+                                    '9950c088-07f8-46ff-b797-62acd2371234'
+                                },
+                  'service_type': 'firewall'
+                  }
+
+    interfaces_rule_info = {'active_provider_mac': 'e1:6d:af:23:b8:91',
+                            'active_stitching_mac': 'e1:6d:af:23:b8:11',
+                            'active_fip': '192.168.100.141',
+                            'service_id':
+                                '6350c0fd-07f8-46ff-b797-62acd2371234',
+                            'tenant_id': '9950c088-07f8-46ff-b797-62acd2371234'
+                            }
+
+    routes = {'context': {},
+              'request_info': {},
+              'vm_mgmt_ip': '127.0.0.1',
+              'service_vendor': 'vyos',
+              'source_cidrs': '11.0.0.0/24',
+              'destination_cidr': '22.0.0.0/24',
+              'gateway_ip': '11.0.0.1',
+              'provider_interface_position': '2',
+              'service_type': 'firewall',
+              }
+
+    healthmonitor = {'context': {},
+                     'request_info': {},
+                     'service_type': 'loadbalancer',
+                     'vmid': '6350c0fd-07f8-46ff-b797-62acd2371234',
+                     'mgmt_ip': '127.0.0.1',
+                     'periodicity': 'initial'
+                     }
+
+"""TestSchemaValidator is a test class to test schema_validator.py using
+   unittest framework
+"""
+
+
+class TestSchemaValidator(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(TestSchemaValidator, self).__init__(*args, **kwargs)
+        self.sv = sv.SchemaValidator()
+        self.sr = SchemaResources()
+
+    def make_request_data(self, resource, kwargs):
+        """Prepares request data
+
+        :param resource - resource_name
+        :param kwargs - kwargs
+
+        Returns: request_data
+
+        """
+
+        request_data = {'info': {'version': 'v1'},
+                        'config': [{'resource': resource,
+                                    'kwargs': kwargs
+                                    }]
+                        }
+        return request_data
+
+    def test_validate_schema_for_request_data(self):
```

Comment:
In the code, validate_schema is not invoked from library users. Our test cases here are validating the validate_schema() function. Do we need to do this? Have we done this as anybody can still use the validate_schema() API in future?